packages/http-server/src/http-server.ts

```diff
@@ -81,7 +114,7 @@ export class HttpServer {
    * @param serverOptions
    */
   constructor(
-    requestListener: RequestListener,
+    requestListener: RequestListener | Http2RequestListener,
```

Comment:
I find this constructor API sub-optimal, because it allows caller to mix HTTP v1 request listener with HTTP v2 server options and vice versa.

Can we use function overloads to implement more tight type checks please?

```ts
class HttpServer {
  // ...
  constructor(
     requestListener: RequestListener,
     serverOptions?: HttpOptions | HttpsOptions
  );

  constructor(
     requestListener: Http2RequestListener,
     serverOptions?: Http2Options | Https2Options
  );

  constructor(
    requestListener: RequestListener | Http2RequestListener,
    serverOptions?: HttpServerOptions,
  ) {
    // the implementation
  }
}
```